contracts/upgradeable_contracts/ForeignOmnibridge.sol

```diff
@@ -141,6 +142,15 @@ contract ForeignOmnibridge is BasicOmnibridge, GasLimitManager {
                 IBurnableMintableERC677Token(_token).safeMint(address(this), _value - balance);
                 balance = _value;
             }
+
+            IInterestImplementation impl = interestImplementation(_token);
+            if (Address.isContract(address(impl))) {
```

Comment:
I believe that the order of these operations does not matter.
Since we do not operate on the actual balance (`token.balanceOf()`) in this piece of code, but rather using recorded value from the storage.

Consider the following situation:
* `20 STAKE` are bridged through the OB. `15 STAKE` of which are invested into EasyStaking, and `5 STAKE` are locked directly on the bridge.
* User requests withdrawal of `30 STAKE`.

First, current code will mint `value - mediatorBalance(STAKE) = 30 STAKE - 20 STAKE = 10 STAKE`, and set local variable `balance = 30 STAKE`
Second, it will see that current available balance is `balance - investedAmount(STAKE) = 30 STAKE - 15 STAKE = 15 STAKE`. Thus, it will withdraw all invested `15 STAKE` tokens from the EasyStaking contract to fulfil the demand.

Reply:
Thanks for clarification. Consider to add this example to the code comments.